plugins/tx_pool/plugin.py

```diff
@@ -0,0 +1,90 @@
+from argparse import (
+    ArgumentParser
+)
+import asyncio
+
+from evm.chains.base import (
+    BaseChain
+)
+from evm.chains.mainnet import (
+    BYZANTIUM_MAINNET_BLOCK,
+    MainnetChain,
+)
+from evm.chains.ropsten import (
+    BYZANTIUM_ROPSTEN_BLOCK,
+    RopstenChain,
+)
+
+from p2p.cancel_token import (
+    CancelToken
+)
+from p2p.peer import (
+    PeerPool
+)
+
+from trinity.extensibility import (
+    BaseActivationEvent,
+    BasePlugin,
+    PluginContext,
+)
+from trinity.extensibility.activation_events import (
+    ResourceAvailableActivationEvent,
+    TrinityStartupActivationEvent,
+)
+from plugins.tx_pool.pool import (
+    TxPool,
+)
+from plugins.tx_pool.validators import (
+    DefaultTransactionValidator
+)
+
+
+class TxPlugin(BasePlugin):
+
+    def __init__(self) -> None:
+        self.peer_pool: PeerPool = None
+        self.cancel_token: CancelToken = None
+        self.chain: BaseChain = None
+        self.is_enabled: bool = None
+
+    @property
+    def name(self) -> str:
+        return "TxPlugin"
+
+    def configure_parser(self, arg_parser: ArgumentParser) -> None:
+        arg_parser.add_argument(
+            "--disable-tx-pool",
+            action="store_true",
+            help="Disable the Transaction Pool",
+        )
+
+    def should_activate(self, activation_event: BaseActivationEvent) -> bool:
+        if isinstance(activation_event, TrinityStartupActivationEvent):
+            self.is_enabled = not activation_event.args.disable_tx_pool
+        if isinstance(activation_event, ResourceAvailableActivationEvent):
+
+            if activation_event.resource_type is PeerPool:
+                self.peer_pool, self.cancel_token = activation_event.resource
```

Comment:
I find it very odd that `should_activate` isn't a pure function, largely based on the name.  I don't love that it has side effects.

Maybe this should be split into two things.

1. `BasePlugin.handle_event`
2. `BasePlugin.should_activate`

That allows for segregation of the mutative code from the logic of whether the plugin should be activated.

Reply:
I like your way of thinking. It didn't feel good to me to mix concerns here either and I very much welcome your suggestion to split this into `handle_event` and `should_activate` APIs.

Btw, I have a somewhat radical™ idea around how to handle these events in an entirely different way but it's probably too radical to get adopted :sweat_smile: Soon™